Backpack-SwiftUI/NavigationTabGroup/Classes/BPKNavigationTabGroup.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A Group of tabs that allows a single tab to be selected at a time.
+public struct BPKNavigationTabGroup: View {
+    private let tabs: [NavigationTabItem]
+    private let style: BPKNavigationTabStyle
+    private let onItemClick: (_ index: Int) -> Void
+    
+    @Binding private var selectedIndex: Int?
+    
+    public init(
+        tabs: [NavigationTabItem],
+        style: BPKNavigationTabStyle = .default,
+        selectedIndex: Binding<Int?>,
```

Comment:
is it really possible to have a nil selection?  as in.. is taht an actual use case that makes sense to have? if not, lets not make this optional, if it is, then this is fine

Reply:
I see on figma there's a design for having nothing selected as an example, but in a real use case I don;t think this makes sense, I'd consider making it non-optional